Source/WebCore/css/CSSComputedStyleDeclaration.cpp

```diff
@@ -105,6 +106,17 @@ const FixedVector<CSSPropertyID>& CSSComputedStyleDeclaration::exposedComputedCS
 
 String CSSComputedStyleDeclaration::getPropertyValue(CSSPropertyID propertyID) const
 {
+    auto canUseShorthandSerializerForPropertyValue = [&]() {
+        switch (propertyID) {
+        case CSSPropertyGridTemplate:
+            return true;
+        default:
+            return false;
+        }
+    };
+    if (isShorthand(propertyID) && canUseShorthandSerializerForPropertyValue())
+        return serializeShorthandValue({ m_element, m_allowVisitedStyle, m_pseudoElementSpecifier }, propertyID);
```

Comment:
I think `ComputedStyleExtractor::valueForPropertyInStyle` is a better place for this, we can also just re-use the already initialized `ComputedStyleExtractor` there too.

Reply:
Looks like we would need to rework some things in order to do this since the serializer uses Strings and the extractor uses CSSValues

Reply:
This is a big problem with our code for computed style — there’s no web platform reason that computed style property values need to be turned into CSS value objects, but there are internal WebKit reasons, like the way we use it in some HTML editing code.

But because the web platform doesn’t specify this, a lot of the CSS values that we create are arbitrary and all that matters from a web platform point of view is that they serialize to a string as specified. At some point we should fix that by eliminating the need to create CSS values from computed style. Then we can delete a ton of redundant computed style code and likely improve the correctness of all computed style shorthand serialization.